contracts/Vault1.sol

```diff
@@ -1,6 +1,49 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
-contract Vault1 {
-    
-}
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract Vault1  {
+    //mapping that the user exists
+    mapping(address => bool) public userExists; 
+
+    //mapping user => balance
+    mapping(address => uint) public balancesOf;
+
+    event Deposit(address user, uint amount);
+    event Withdraw(address owner, uint amount);
+
+    ERC20 public token;
```

Comment:
You can actually use the interface here instead of the whole implementation. 
So the import would be `import "@openzeppelin/contracts/token/ERC20/IERC20.sol";`
and here it would be `IERC20 public token;`